gcovr/templates/source_page.html

```diff
@@ -108,6 +125,40 @@
         </details>
         {% endif %}
       </td>
+      {% if SHOW_DECISION %}
+      <td class="linedecision">
+        {% if row.linedecision %}
+        <details class="linedecisionDetails">
+          <summary class="linedecisionSummary">{{row.linedecision.taken}}/{{row.linedecision.total}}</summary>
+          <div class="linedecisionContents">
+            {% for decision in row.linedecision.decisions %}
+            {% if decision.uncheckable == False %}
+            {% if decision.taken %}
+            {% if loop.index0 % 2 == 0 %}
+            <div class="takenDecision">&check; Decision 'true' taken {{decision.count}} times.</div>
+            {% else %}
+            <div class="takenDecision">&check; Decision 'false' taken {{decision.count}} times.</div>
+            {% endif %}
+            {% else %}
+            {% if loop.index0 % 2 == 0 %}
+            <div class="notTakenDecision">&cross; Decision 'true' not taken.</div>
+            {% else %}
+            <div class="notTakenDecision">&cross; Decision 'false' not taken.</div>
+            {% endif %}
+            {% endif %}
+            {% else %}
+            {% if loop.index0 % 2 == 0 %}
+            <div class="uncheckedDecision">? Decision 'true' couldn't be analyzed.</div>
+            {% else %}
+            <div class="uncheckedDecision">? Decision 'false' couldn't be analyzed.</div>
+            {% endif %}
+            {% endif%}
+            {% endfor %}
```

Comment:
There's a lot of business logic going on here. Could some of it be moved into the Python code? At least, the decision name (true/false) could be handled outside of the template.

Reply:
I'll also invert the if, I hate a negated if with an else branch.
```
if xxx == False
   ...
else
   ...
endif
```